missing notifications

<!--
This is a bug report tracker. New features are discussed in the forum: https://support.delta.chat

Please fill out as much of this form as you can (leaving out stuff that is not applicable is ok).
-->

- Platform (android/blackberry/anbox): Android 7.0
- Device: Huawei nova
- Delta Chat Version: 0.500.0
- Expected behavior: notification for new message
- Actual behavior: no notification
- Steps to reproduce the problem:
- Screenshots:
- Logs: I will send a logfile to @r10s

<!--
Debug logs can be copied from within the Delta Chat app with
Settings menu -> Advanced -> Debug log

Alternatively from the Android system log:
`adb logcat -v time -s DeltaChat`

Logs may contain private data 
which shall be removed or anonymised prior to posting.
-->

log from another user:
[user2.txt](https://github.com/deltachat/deltachat-android/files/3359984/user2.txt)

interesting part:
```
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder: java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at com.bumptech.glide.request.RequestFutureTarget.doGet(RequestFutureTarget.java:212)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at com.bumptech.glide.request.RequestFutureTarget.get(RequestFutureTarget.java:115)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.SingleRecipientNotificationBuilder.setChat(SingleRecipientNotificationBuilder.java:78)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.MessageNotifier.createSingleChatNotification(MessageNotifier.java:243)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.MessageNotifier.sendSingleChatNotification(MessageNotifier.java:212)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.MessageNotifier.sendNotifications(MessageNotifier.java:188)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.MessageNotifier.updateNotification(MessageNotifier.java:104)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.MessageNotifierCompat.lambda$updateNotification$0$MessageNotifierCompat(MessageNotifierCompat.java:34)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.MessageNotifierCompat$$Lambda$0.run(Unknown Source)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.util.Util.runOnAnyBackgroundThread(Util.java:331)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.notifications.MessageNotifierCompat.updateNotification(MessageNotifierCompat.java:34)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at org.thoughtcrime.securesms.ApplicationContext$1.handleEvent(ApplicationContext.java:100)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at com.b44t.messenger.DcEventCenter.lambda$sendToObservers$1$DcEventCenter(DcEventCenter.java:77)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at com.b44t.messenger.DcEventCenter$$Lambda$1.run(Unknown Source)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder:     at java.lang.Thread.run(Thread.java:761)
07-04 11:23:27.678 22914 29571 W SingleRecipientNotificationBuilder: Caused by: com.bumptech.glide.load.engine.GlideException: Failed to load resource
07-04 11:23:27.685 22914 29571 W NotificationState: Added chat: 42
07-04 11:23:27.685 22914 29571 W NotificationState: Added chat: 42
07-04 11:23:27.747 22914 29571 D MediaPlayer: setSubtitleAnchor in MediaPlayer
07-04 11:23:27.747 22914 22914 D MediaPlayer: setSubtitleAnchor in MediaPlayer
07-04 11:23:27.751 22914 29571 D Ringtone: Successfully created local player
```

--- a/src/org/thoughtcrime/securesms/notifications/MessageNotifier.java
+++ b/src/org/thoughtcrime/securesms/notifications/MessageNotifier.java
@@ -3,6 +3,7 @@ package org.thoughtcrime.securesms.notifications;
 import android.app.NotificationManager;
 import android.content.Context;
 import android.media.AudioManager;
+import android.media.Ringtone;
 import android.media.RingtoneManager;
 import android.media.SoundPool;
 import android.net.Uri;
@@ -279,13 +280,18 @@ abstract class MessageNotifier {
 
     }
 
-    void playNotificationSound(Uri ringtone, boolean vibrate) {
-        RingtoneManager.getRingtone(appContext, ringtone).play();
-        if (vibrate) {
+    private void playNotificationSound(Uri uri, boolean vibrate) {
+        Ringtone ringtone = RingtoneManager.getRingtone(appContext, uri);
+        if (ringtone!=null) {
+            ringtone.play();
+        }
 
+        if (vibrate) {
             Vibrator v = (Vibrator) appContext.getSystemService(Context.VIBRATOR_SERVICE);
-            v.vibrate(100);
-            v.vibrate(200);
+            if (v!=null) {
+                v.vibrate(100);
+                v.vibrate(200);
+            }
         }
     }
 

